Lib/_strptime.py

```diff
@@ -470,7 +470,10 @@ def _strptime(data_string, format="%a %b %d %H:%M:%S %Y"):
                 minutes = int(z[3:5])
                 seconds = int(z[5:7] or 0)
                 gmtoff = (hours * 60 * 60) + (minutes * 60) + seconds
-                gmtoff_fraction = int(z[8:] or 0)
+                gmtoff_remainder = z[8:]
+                # Pad to always return microseconds.
+                gmtoff_remainder_padding = "0" * (6 - len(gmtoff_remainder))
```

Comment:
I did a microbenchmark for this vs. multiplying by `10**(6-len(gmtoff_remainder))` and it seems like your version is either faster or the same speed. Maybe that's why you chose this method, but I thought I'd chime in since I went through the trouble to check it.